inc/functions.php

```diff
@@ -289,15 +293,60 @@ function colormag_favicon() {
 
 /****************************************************************************************/
 
+/**
+ * Change hex code to RGB
+ * Source: https://css-tricks.com/snippets/php/convert-hex-to-rgb/#comment-1052011
+ */
+function colormag_hex2rgb( $hexstr ) {
+    $int = hexdec( $hexstr );
+    $rgb = array( "red" => 0xFF & ( $int >> 0x10 ), "green" => 0xFF & ( $int >> 0x8 ), "blue" => 0xFF & $int );
+    $r = $rgb[ 'red' ];
+    $g = $rgb[ 'green' ];
+    $b = $rgb[ 'blue' ];
+
+    return "rgba( $r, $g, $b, 0.85 )";
+}
+
+/**
+ * Generate darker color
+ * Source: http://stackoverflow.com/questions/3512311/how-to-generate-lighter-darker-color-with-php
+ */
+function colormag_darkcolor( $hex, $steps ) {
+	// Steps should be between -255 and 255. Negative = darker, positive = lighter
+	$steps = max( -255, min( 255, $steps ) );
+
+	// Normalize into a six character long hex string
+	$hex = str_replace( '#', '', $hex );
+	if ( strlen( $hex ) == 3 ) {
+		$hex = str_repeat( substr( $hex, 0, 1 ), 2 ) . str_repeat( substr( $hex, 1, 1 ), 2 ) . str_repeat( substr( $hex, 2, 1 ), 2 );
+	}
+
+	// Split into three parts: R, G and B
+	$color_parts = str_split( $hex, 2 );
+	$return = '#';
+
+	foreach ( $color_parts as $color ) {
+		$color   = hexdec( $color ); // Convert to decimal
+		$color   = max( 0, min( 255, $color + $steps ) ); // Adjust color
+		$return .= str_pad( dechex( $color ), 2, '0', STR_PAD_LEFT ); // Make two char hex code
+	}
+
+	return $return;
+}
+
+/****************************************************************************************/
+
 add_action('wp_head', 'colormag_custom_css');
 /**
  * Hooks the Custom Internal CSS to head section
  */
 function colormag_custom_css() {
 	$colormag_internal_css = '';
 	$primary_color = get_theme_mod( 'colormag_primary_color', '#289dcc' );
+	$primary_opacity = colormag_hex2rgb( $primary_color );
```

Comment:
@rabindratharu Could not see this variable is being used anywhere. So, it it really needed?